WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -3056,59 +3021,23 @@ function WeakAuras.ReloadTriggerOptions(data)
       WeakAuras.ReloadTriggerOptions(data);
     end
 
-    removeFuncs(displayOptions[id]);
-
+    trigger_options = union(trigger_options, WeakAuras.GetGenericTriggerOptions(data));
     if(optionTriggerChoices[id] >= 1 and getAll(data, {"trigger", "type"}) == "aura") then
-      local aura_options = WeakAuras.GetBuffTriggerOptions(data, trigger);
-      displayOptions[id].args.trigger.args = union(trigger_options, aura_options);
-      removeFuncs(displayOptions[id].args.trigger);
-      displayOptions[id].args.trigger.args.type.set = options_set;
+      local aura_options = WeakAuras.GetBuffTriggerOptions(data);
+      trigger_options = union(trigger_options, aura_options);
     elseif(optionTriggerChoices[id] >= 1 and (getAll(data, {"trigger", "type"}) == "event" or getAll(data, {"trigger", "type"}) == "status")) then
       local event = getAll(data, {"trigger", "event"});
       local unevent = getAll(data, {"trigger", "unevent"});
       if(event and WeakAuras.event_prototypes[event]) then
-        local trigger_options_created;
```

Comment:
This code block fetched subeventPrefix and subeventSuffix. But those arguments to ConstructFunction were ignored anyway, so that whole code block was unnecessary.